drafts/web-editor-monaco-with-language-server-support.md

```diff
@@ -0,0 +1,971 @@
+---
+title: Create a custom web editor using Monaco and Language Server Protocol (LSP)
+domain: software-engineering-corner.hashnode.dev
+tags: Web Editor, Monaco, Monaco Language Client, Language Server, LSP
+cover: https://cdn.hashnode.com/res/hashnode/image/upload/v1719718605895/bv2LfyFWS.jpg?auto=format
+publishAs: cjayashantha, NipunaMarcusZuhlke
+hideFromHashnodeCommunity: false
+--- 
+
+# Create a custom web editor using Monaco and Language Server Protocol (LSP)
+
+If you are developing a domain-specific language or custom language, chances are high that current editors might not support features such as syntax highlighting, diagnostics, and code autocompletion. To enable these features, you need a Language Server Protocol (LSP) and sometimes a custom editor. In this article, we will discuss how to develop a web editor and connect it to a language server using the WebSocket protocol.
```

Comment:
Here you use `autocompletion`, while you use `auto-completion` everywhere else in the article. Probably would try to be consistent

Reply:
Maybe from a readers perspective it would easier to follow if you show the screenshort/gif that we should at the end already here. Then it would be super clear what we are aiming for through the articles

Reply:
Would still vote for showing the [final screen](https://github.com/Zuehlke/software-engineering-corner/pull/31/files#diff-5ddfe36215164246e909db4ba17080865f69c8de7c209b9af503d93106cf2f8eR415) upfront in the first article to show what we want to achieve 😄